### PR TITLE
DBTP-277 ./copilot-helper.py check-cloudformation security

### PR DESCRIPTION
**Pull request by gabelton**
https://uktrade.atlassian.net/browse/DBTP-277

Adds [checkov ](https://pypi.org/project/checkov/) as part of new `static` subcommand

### CODE REVIEW

File: dbt_copilot_helper/commands/check_cloudformation.py [hunk 1]
```diff
@@ -49,6 +50,25 @@ def lint(ctx: click.Context, directory: str) -> bool:
     return success
 
 
+@check_cloudformation.command()
+@click.option("-d", "--directory", type=str, default="copilot")
+@click.pass_context
+def static(ctx: click.Context, directory: str) -> bool:
```
Comment (WillGibson) [inline dbt_copilot_helper/commands/check_cloudformation.py:56]: What does `static` mean in this context?

File: buildspec.test.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ phases:
       - echo Setting local Python versions
       - pyenv versions | awk 'match($0, /\d\.\d+\.\d+/) { print substr($0, RSTART, RLENGTH) }' > .python-version
       - echo Installing dependencies
+      - pip install --upgrade pip
```
Comment (yusufsheiqh) [inline buildspec.test.yml:9]: Why was this needed btw?

File: dbt_copilot_helper/commands/check_cloudformation.py [hunk 3]
```diff
@@ -49,6 +50,25 @@ def lint(ctx: click.Context, directory: str) -> bool:
...
+    addons_manifests = f"{directory}/**/addons/*.yml"
```
Comment (WillGibson) [inline dbt_copilot_helper/commands/check_cloudformation.py:57]: Would be nice to have a description for the documentation, and I think we should inclue the `overrides` directories too...
```suggestion
    """Scans generated CloudFormation templates for security issues."""
    addons_manifests = f"{directory}/**/addons/*.yml"
    addons_manifests = f"{directory}/**/overrides/*.yml"
```

File: tests/copilot_helper/fixtures/insecure_cloudformation_template.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Resources:
+  MyDB1:
+    Type: 'AWS::RDS::DBInstance'
+    Properties:
+      DBName: 'mydb'
+      DBInstanceClass: 'db.t3.micro'
+      Engine: 'mysql'
+      MasterUsername: 'master'
+      MasterUserPassword: 'password'
+      StorageEncrypted: false
```
Comment (WillGibson) [inline tests/copilot_helper/fixtures/insecure_cloudformation_template.yml:10]: Might be worth adding a comment explaining how this template is insecure.

File: buildspec.test.yml (hunk 2 above)
Comment (gabelton) [reply buildspec.test.yml:9]: This was my (failed) attempt to fix the failing build. Will remove in a sec

File: dbt_copilot_helper/commands/check_cloudformation.py (hunk 1 above)
Comment (WillGibson) [reply dbt_copilot_helper/commands/check_cloudformation.py:56]: Imagining a rename of `check-cloudformation` imminently and the whole thing, verb approach, maybe...

`copilot-helper cloudformation check-security`?

File: dbt_copilot_helper/commands/check_cloudformation.py (hunk 3 above)
Comment (gabelton) [reply dbt_copilot_helper/commands/check_cloudformation.py:57]: @WillGibson cfn.patches.yml isn't a cloudformation template itself though is it? The `lint` command doesn't check it

Comment (WillGibson) [reply dbt_copilot_helper/commands/check_cloudformation.py:57]: You're right, they are just YAML patches. Ignore this comment.

**Review by WillGibson — APPROVED** [review]
